src/library/scala/util/Random.scala

```diff
@@ -70,6 +70,16 @@ class Random(val self: java.util.Random) extends AnyRef with Serializable {
    */
   def nextLong(): Long = self.nextLong()
 
+  def nextLong(n: Long): Long = {
+    if (n <= 0) throw new java.lang.IllegalArgumentException("bound must be positive")
+    if (n <= Int.MaxValue) nextInt(n.toInt).toLong
+    else {
+      val RAND_MAX = Long.MaxValue     // 2^63-1
+      val x = nextLong() >>> 1         // [0, 2^63), strip the sign bit
+      if (x < (RAND_MAX - (RAND_MAX % n))) x % n else nextLong(n)
```

Comment:
Heh, ok. I'll try to find some time to learn about it more. I don't use scala.js so it'll be a series of new experiences.

Anyway - if you're comfortable with this code, awesome. I don't know why it's failing an additional test (I know @SethTisue is looking into the MiMa failures) but hopefully it's not related to any code breakage with the new commit.